src/Microsoft.Azure.EventHubs/Amqp/AmqpMessageConverter.cs

```diff
@@ -235,31 +234,22 @@ private static void UpdateEventDataHeaderAndProperties(AmqpMessage amqpMessage,
                     }
                 }
             }
-
-            //if ((sections & SectionFlag.Properties) != 0)
-            //{
-            //    var properties = amqpMessage.Properties;
-            //    AddIfTrue(data.SystemProperties, properties, p => p.MessageId != null, Properties.MessageIdName, p => p.MessageId.ToString());
-            //    AddIfTrue(data.SystemProperties, properties, p => p.UserId.Array != null, Properties.UserIdName, p => p.UserId);
-            //    AddIfTrue(data.SystemProperties, properties, p => p.To != null, Properties.ToName, p => p.To.ToString());
-            //    AddIfTrue(data.SystemProperties, properties, p => p.Subject != null, Properties.SubjectName, p => p.Subject);
-            //    AddIfTrue(data.SystemProperties, properties, p => p.ReplyTo != null, Properties.ReplyToName, p => p.ReplyTo.ToString());
-            //    AddIfTrue(data.SystemProperties, properties, p => p.CorrelationId != null, Properties.CorrelationIdName, p => p.CorrelationId.ToString());
-            //    AddIfTrue(data.SystemProperties, properties, p => p.ContentType.Value != null, Properties.ContentTypeName, p => p.ContentType.ToString());
-            //    AddIfTrue(data.SystemProperties, properties, p => p.ContentEncoding.Value != null, Properties.ContentEncodingName, p => p.ContentEncoding.ToString());
-            //    AddIfTrue(data.SystemProperties, properties, p => p.AbsoluteExpiryTime != null, Properties.AbsoluteExpiryTimeName, p => p.AbsoluteExpiryTime);
-            //    AddIfTrue(data.SystemProperties, properties, p => p.CreationTime != null, Properties.CreationTimeName, p => p.CreationTime);
-            //    AddIfTrue(data.SystemProperties, properties, p => p.GroupId != null, Properties.GroupIdName, p => p.GroupId);
-            //    AddIfTrue(data.SystemProperties, properties, p => p.GroupSequence != null, Properties.GroupSequenceName, p => p.GroupSequence);
-            //    AddIfTrue(data.SystemProperties, properties, p => p.ReplyToGroupId != null, Properties.ReplyToGroupIdName, p => p.ReplyToGroupId);
-            //}
         }
 
         static ArraySegment<byte> StreamToBytes(Stream stream)
         {
-            MemoryStream memoryStream = new MemoryStream(512);
-            stream.CopyTo(memoryStream, 512);
-            ArraySegment<byte> buffer = new ArraySegment<byte>(memoryStream.ToArray());
+            ArraySegment<byte> buffer;
+
+            if (stream == null)
```

Comment:
>(stream == null) [](start = 15, length = 16)

In the zero bytes payload case stream is null. So doing a null check here fixes NullRef.